src/core/Akka/Event/LoggingBus.cs

```diff
@@ -164,46 +173,68 @@ internal void StopDefaultLoggers(ActorSystem system)
+            _shutdownCts.Cancel(false);
+            _shutdownCts.Dispose();
+            
+            // Stop all currently running loggers
             foreach (var logger in _loggers)
             {
-                if (!(logger is MinimalLogger))
-                {
-                    Unsubscribe(logger);
-                    if (logger is IInternalActorRef internalActorRef)
-                    {
-                        internalActorRef.Stop();
-                    }
-                }
+                RemoveLogger(logger);
             }
 
             Publish(new Debug(SimpleName(this), GetType(), "All default loggers stopped"));
         }
 
-        private void AddLogger(ActorSystemImpl system, Type loggerType, LogLevel logLevel, string loggingBusName, TimeSpan timeout)
+        private void RemoveLogger(IActorRef logger)
         {
-            var loggerName = CreateLoggerName(loggerType);
-            var logger = system.SystemActorOf(Props.Create(loggerType).WithDispatcher(system.Settings.LoggersDispatcher), loggerName);
-            var askTask = logger.Ask(new InitializeLogger(this), timeout);
-
-            object response = null;
-            try
+            if (!(logger is MinimalLogger))
             {
-                response = askTask.Result;
-            }
-            catch (Exception ex) when (ex is TaskCanceledException || ex is AskTimeoutException)
-            {
-                Publish(new Warning(loggingBusName, GetType(),
-                     string.Format("Logger {0} [{2}] did not respond within {1} to InitializeLogger(bus)", loggerName, timeout, loggerType.FullName)));
+                Unsubscribe(logger);
+                if (logger is IInternalActorRef internalActorRef)
+                {
+                    internalActorRef.Stop();
+                }
             }
-                
-            if (!(response is LoggerInitialized))
-                throw new LoggerInitializationException($"Logger {loggerName} [{loggerType.FullName}] did not respond with LoggerInitialized, sent instead {response}");
-            
+        }
 
-            _loggers.Add(logger);
-            SubscribeLogLevelAndAbove(logLevel, logger);
-            Publish(new Debug(loggingBusName, GetType(), $"Logger {loggerName} [{loggerType.Name}] started"));
+        private void AddLogger(ActorSystemImpl system, Type loggerType, string loggingBusName)
+        {
+            var loggerName = CreateLoggerName(loggerType);
+            var logger = system.SystemActorOf(Props.Create(loggerType).WithDispatcher(system.Settings.LoggersDispatcher), loggerName);
+            var askTask = logger.Ask(new InitializeLogger(this), Timeout.InfiniteTimeSpan, _shutdownCts.Token);
             
+            _startupState[askTask] = new LoggerStartInfo(loggerName, loggerType, logger);
+            askTask.ContinueWith(t =>
+                {
+                    var info = _startupState[t];
```

Comment:
Not safe to do this in a continuation, since this is now shared mutable state.

Reply:
I'd keep the rest of the continuation but basically just purge the `_startupState` after the `Task.WaitAll`.

Reply:
It should be safe since this is the only class where it is modified and there is no cross-pollination, each task handles its own continuation and have their own Task as their key.
We had to held on to `_startupState` because of the possibility that some loggers are not ready after the `Task.WaitAll` and they needed to retrieve their info. Would it be safe to use closure instead?

Reply:
Removed the dictionary, changed it to closure. Just needed the logger name for logging purposes outside of the Task continuation